lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,230 @@
+            url == "https://aka.ms/deprecateLicenseUrl"
+          end
+
+          def text_content_url(url)
+            # Convert github file URLs to raw URLs
+            return url unless match = url.match(/https?:\/\/(?:www\.)?github.com\/([^\/]+)\/([^\/]+)\/blob\/(.*)/i)
+            "https://github.com/#{match[1]}/#{match[2]}/raw/#{match[3]}"
+          end
+
+          def retrieve_license(url)
+            return unless url
+            return if ignored_url?(url)
+
+            # Transform URLs that are known to return HTML but have a corresponding text-based URL
+            text_url = text_content_url(url)
+
+            raw_content = fetch_content(text_url)
+            strip_html(raw_content)
+          end
+
+          def fetch_content(url, redirect_limit = 5)
+            url = URI.parse(url) if url.instance_of? String
+            return @response_by_url[url] if (@response_by_url ||= {}).key?(url)
+            return if redirect_limit == 0
+
+            begin
+              response = Net::HTTP.get_response(url)
+              case response
+              when Net::HTTPSuccess     then
+                @response_by_url[url] = response.body
+              when Net::HTTPRedirection then
+                redirect_url = URI.parse(response["location"])
+                if redirect_url.relative?
+                  redirect_url = url + redirect_url
+                end
+                # The redirect might be to a URL that requires transformation, i.e. a github file
+                redirect_url = text_content_url(redirect_url.to_s)
+                @response_by_url[url] = fetch_content(redirect_url, redirect_limit - 1)
+              end
+            rescue => error
+              # Host might no longer exist or some other error, ignore
+            end
+          end
+        end
+      end
+
+      def enabled?
+        nuget_config_exists(config.pwd)
+      end
+
+      def nuget_obj_root
+        config.dig("nuget", "obj_root") || config.pwd
+      end
+
+      def dump_projects?
+        config.dig("nuget", "projects", "dump")
+      end
+
+      def exclude_projects
+        config.dig("nuget", "projects", "exclude") || []
+      end
+
+      def excluded_project?(project_name)
+        exclude_projects.any? do |pattern|
+          File.fnmatch?(pattern, project_name, File::FNM_PATHNAME | File::FNM_CASEFOLD)
+        end
+      end
+
+      def nuget_config_exists(root)
+        # Multiple supported casings: https://github.com/NuGet/Home/issues/1427
+        File.exist?(File.join(root, "nuget.config")) || File.exist?(File.join(root, "NuGet.config")) || File.exist?(File.join(root, "NuGet.Config"))
+      end
+
+      def enumerate_dependencies
+        packages.map do |key, package|
+          metadata = {
+            "type" => NuGet.type,
+            "name" => package["name"]
+          }
+
+          # Emit the names of the projects that consume a particular dependency.
```

Comment:
👍 being able to see how a package is included in a project sounds really nice, but I don't understand this comment - how does this tie back to excluding projects?

I'm a little concerned that exclusions are applied to not evaluate nested dependencies, which would be problematic if the excluded dependency was updated to bring in a new dependency that should not be excluded.  

Dependency sources should always evaluate the full dependency tree, with the `ignore` setting used to skip further operations (cache, status, list) on individual dependencies.

Reply:
See above comment. This isn't about transitive dependencies - it's about excluding [test | non-shipping] projects from pulling in root dependencies.

Reply:
Per our discussion, removed this and at least for now will only support targeting 1 project at a time, where the `source_path` must be the directory containing the `project.assets.json` file.